src/inflators/particle-emitter.ts

```diff
@@ -0,0 +1,89 @@
+import { addComponent } from "bitecs";
+import { addObject3DComponent } from "../utils/jsx-entity";
+import { ParticleEmitterTag } from "../bit-components";
+import { HubsWorld } from "../app";
+import defaultSrcImage from "../assets/images/warning_icon.png";
+import { ParticleEmitter } from "lib-hubs/packages/three-particle-emitter/lib/esm/index";
+
+const defaultSrcUrl = new URL(defaultSrcImage, window.location.href).href;
+
+export type ParticleEmitterParams = {
+  src: string;
+  startColor: string;
+  middleColor: string;
+  endColor: string;
+  startOpacity: number;
+  middleOpacity: number;
+  endOpacity: number;
+  colorCurve: string;
+  sizeCurve: string;
+  startSize: number;
+  endSize: number;
+  sizeRandomness: number;
+  ageRandomness: number;
+  lifetime: number;
+  lifetimeRandomness: number;
+  particleCount: number;
+  startVelocity: [number, number];
+  endVelocity: [number, number, number];
+  velocityCurve: string;
+  angularVelocity: number;
+};
+
+const DEFAULTS = {
+  src: defaultSrcImage,
```

Comment:
I wan to get it so that `inflateParticleEmitter` takes a texture instead of a `src` (similar to how `inflateImage` takes a texture instead of a src). Since the gltf-exposed component expects a src we made a seperate loader for `image` in gltf called `inflateImageLoader`, that ends up using a MediaLoader.

Not sure its worth trying to do that here, might make more sense to just try and design it that we when we make the GPUParticleEmitter component and deprecate this one.

Reply:
Yeah, that makes sense. It will also make sense in the future to allow direct texture references in glTF components like `image` or `particle-emitter` similarly to what we do for other properties like `envMap`. Probably even allow both paths, URLs and texture refs.

I'm going to leave it as it is for now as I'll like to do a pass after the migration an add support for a more modern particle system.